tests/ops/qubit/test_parametric_ops.py

```diff
@@ -2780,13 +2786,20 @@ def test_PauliRot_all_Identity_broadcasted(self):
         op = qml.PauliRot(theta, "II", wires=[0, 1])
         decomp_ops = op.decomposition()
 
-        phases = np.exp(-1j * theta / 2)
-        assert np.allclose(op.eigvals(), np.outer(phases, np.ones(4)))
-        mat = op.matrix()
-        for phase, sub_mat in zip(phases, mat):
-            assert np.allclose(sub_mat, phase * np.eye(4))
+        assert len(decomp_ops) == 1
+
+        decomp_op = decomp_ops[0]
+
+        assert decomp_op.name == "GlobalPhase"
+
+        # global phase acts on all wires so wire attribute is unused
+        assert decomp_op.wires == Wires([])
+
+        for param, angle in zip(decomp_op.data[0], theta):
+            assert qml.math.allclose(param, angle / 2)
 
-        assert len(decomp_ops) == 0
+        for op_matrix, decomp_phase in zip(op.matrix(), decomp_op.matrix().T):
```

Comment:
```suggestion
        assert len(op) == len(decomp_op)
        for op_matrix, decomp_phase in zip(op.matrix(), decomp_op.matrix().T):
```

Reply:
But op is always going to be a single op right? So it does not even have a length

Reply:
Lol my bad. Why are you iterating over elements then? Simply use allclose for equality.

Reply:
I can add the check for it like:
`assert len(op.matrix()) == len(decomp_op.matrix())`